Sources/OpenAI/AIProxy/AIProxyCertificatePinning.swift

```diff
@@ -0,0 +1,181 @@
+//
+//  AIProxyCertificatePinning.swift
+//
+//
+//  Created by Lou Zell on 6/23/24.
+//
+
+import Foundation
+import OSLog
+
+private let aiproxyLogger = Logger(subsystem: Bundle.main.bundleIdentifier ?? "UnknownApp",
+                                   category: "SwiftOpenAI+AIProxyCertificatePinning")
+
+/// ## About
+/// Use this class in conjunction with a URLSession to adopt certificate pinning in your app.
+/// Cert pinning greatly reduces the ability for an attacker to snoop on your traffic.
+///
+/// A common misunderstanding about https is that it's hard for an attacker to read your traffic.
+/// Unfortunately, that is only true if you, as the developer, control both sides of the pipe.
+/// As an app developer, this is almost never the case. You ship your apps to the app store, and
+/// attackers install them. When an attacker has your app on hardware they control (e.g. an iPhone),
+/// it is trivial for them to MITM your app and read encrypted traffic.
+///
+/// Certificate pinning adds an additional layer of security by only allowing the TLS handshake to
+/// succeed if your app recognizes the public key from the other side. I have baked in several AIProxy
+/// public keys to this implementation.
+///
+/// This also functions as a reference implementation for any other libraries that want to interact
+/// with the aiproxy.pro service using certificate pinning.
+///
+/// ## Implementor's note, and a gotcha
+/// Use an instance of this class as the delegate to URLSession. For example:
+///
+///     let mySession = URLSession(
+///        configuration: .default,
+///        delegate: AIProxyCertificatePinningDelegate(),
+///        delegateQueue: nil
+///     )
+///
+/// In a perfect world, this would be all that is required of you. In fact, it is all that is required to protect requests made
+/// with `await mySession.data(for:)`, because Foundation calls `urlSession:didReceiveChallenge:`
+/// internally. However, `await mySession.bytes(for:)` is not protected, which is rather odd. As a workaround,
+/// change your callsites from:
+///
+///     await mySession.bytes(for: request)
+///
+/// to:
+///
+///     await mySession.bytes(
+///         for: request,
+///         delegate: mySession.delegate as? URLSessionTaskDelegate
+///     )
+///
```

Comment:
@jamesrochabrun you will see I made the above changes to two places in the core of this lib. I tested both of them using the `SwiftOpenAIExample` app under the 'Default OpenAI Service' tab